Source/Parallelization/WarpXComm.cpp

```diff
@@ -171,136 +171,191 @@ WarpX::UpdateAuxilaryDataStagToNodal ()
 
         // Bfield
         {
```

Comment:
I will add some more comments later on this file.

Reply:
As a follow-up PR: we could avoid the duplication for E and B, and instead have a function that is generic.

Reply:
Ported to #5078, we will fix this in a separate PR.